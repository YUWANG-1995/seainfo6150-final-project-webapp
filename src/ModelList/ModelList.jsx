import React, { useState, useEffect} from "react";
import { Link } from "react-router-dom";
import { isEmpty } from "lodash";
import ModelListItem from "../ModelListItem/ModelListItem";
import styles from "./ModelList.module.css";

const ModelList = (props) => {
  const [fetchedModels, setFetchedModels] = useState({});

  useEffect( () => {
    const fetchModels = async () => {
      const result = await fetch("http://demo1468395.mockable.io/brands");
      const response = await result.json()
      console.log(JSON.stringify(Object.values(response['BMW']['model'])));
      setFetchedModels(response);
    };

    if(isEmpty(fetchedModels)) {
      fetchModels();
    }
  }, [fetchedModels]);

  let displayAllModels;
  if(!isEmpty(fetchedModels)){
    let oneBrand = fetchedModels[props.brand]['model'];
    displayAllModels = (
      <div>
        <ul className={styles.modelList}>
          {Object.values(oneBrand).map((model, index) => (
            <Link key={index} to={"/cars/"+props.brand+"/"+model.modelId}>
              <li key={index}>
                <ModelListItem model={model} key={index} />
              </li>
            </Link>
          ))}
        </ul>
      </div>
    );
  } else {
    displayAllModels = null;
  }

  return (
    displayAllModels
  );
}
export default ModelList;
