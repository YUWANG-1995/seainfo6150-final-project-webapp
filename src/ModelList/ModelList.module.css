.container{

}

.modelList{
  display: grid;
  grid-template-columns: 1fr 1fr;
  column-gap: 20px;
  list-style: none;
}

@media (max-width: 1000px){
  .modelList{
    grid-template-columns: 1fr;
    column-gap: 30px;
  }
}

@media (min-width: 1000px){
  .modelList{
    grid-template-columns: 1fr 1fr;
    column-gap: 20px;
  }
}
