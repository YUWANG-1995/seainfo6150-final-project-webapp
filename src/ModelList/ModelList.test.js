import React from "react";
import { render } from "@testing-library/react";
import ModelList from "./ModelList.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the ModelList Component Correctly", () => {
  const test = [
    {
      brand: 'BMW',
      model: {
        "X1": {
          modelId: "X1"
        },
        "X4": {
          modelId: "X4"
        }
      },
      brand: "Audi",
      model: {
        "A1": {
          modelId: "A1"
        },
        "A3": {
          modelId: "A3"
        }
      }
    }
  ]

  const { container } = render(
    <BrowserRouter>
      <ModelList
        brands={Object.values(test)}
      />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
