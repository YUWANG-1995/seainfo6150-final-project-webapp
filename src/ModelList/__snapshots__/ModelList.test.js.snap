// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ModelList Component Correctly 1`] = `<div />`;
